src/Compilers/Core/Portable/SourceGeneration/RunResults.cs

```diff
@@ -70,6 +70,17 @@ public ImmutableArray<SyntaxTree> GeneratedTrees
                 return _lazyGeneratedTrees;
             }
         }
+
+        public sealed class CancelledResult : GeneratorDriverRunResult
+        {
+            internal CancelledResult(TimeSpan elapsedTime, GeneratorRunResult? cancelledOn)
+                : base(ImmutableArray<GeneratorRunResult>.Empty, elapsedTime)
+            {
+                LastGeneratorRunning = cancelledOn;
+            }
+
+            public GeneratorRunResult? LastGeneratorRunning { get; }
```

Comment:
Currently we'll always have a value, because we only return a cancelled result if the cancellation occurred during a generator run, but I didn't want to box us in to graceful cancellation at other points too. If we go with the array, then its easier as we can just return an empty one in that case.